### PR TITLE
Fix scripts running tests [ECR-1993]:

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview
The logic enabling tests has been moved to Maven build definitions,
therefore, there is no need to set the global user property
that will disable *all* tests (see 5e2055d).

Also, when compiling artefacts in run_native_integration_tests.sh,
there is no need to pass -DskipTests.

---

### Definition of Done

- [x] System libsodium is used by EJB-native
- [x] Libsodium is installed
- [x] There are no TODOs left in the code
- [x] The [continuous integration build](https://www.travis-ci.org/exonum/exonum-java-binding) passes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -46,9 +50,8 @@ before_install:
   - cargo-audit -V || cargo install cargo-audit --force
     # List all installed cargo packages.
   - cargo install --list
-    # libjava_bindings.so and native application both need to load common Rust libs (eg libstd.so).
-  - export LD_LIBRARY_PATH=$RUST_LIB_DIR:$LD_LIBRARY_PATH
-    # force building instead of using from apt.
+    # Force building Exonum dependencies instead of using the system ones, if any.
+    # Required because using system results in broken artefacts on older CPUs, see ECR-1658.
   - export SODIUM_BUILD=1
```
Comment (alexander-irbis) [inline .travis.yml]: If we decided to install system libsodium.so for Java, should we reuse it in Rust part too?
`export SODIUM_LIB_DIR="/usr/lib"`

Comment (vitvakatu) [reply .travis.yml]: We had crashes with libsodium from repos, remember?

Comment (dmitry-timofeev) [reply .travis.yml]: I noticed that we build libsodium but couldn't remember _why_ we do. Still can't remember the crashes :man_shrugging: 

#399 might make system libsodium unnecessary

Comment (vitvakatu) [reply .travis.yml]: afaik we had some problems (not exactly crashes), so @alexander-irbis found that building libsodium from source solves these problems.

Comment (dmitry-timofeev) [reply .travis.yml]: I remember RocksDB & SIGILL on some older CPUs.

Comment (vitvakatu) [reply .travis.yml]: :thinking: interesting, so why we build libsodium?

Comment (alexander-irbis) [reply .travis.yml]: > We had crashes with libsodium from repos, remember?

Definitely that issue with SIGILL was discussed primarily in the context of RocksDB.
LIbsodium was switched to build at the same time with RocksDB but for another reason, as try to fix crashing issue on Linux. It did not help.

Comment (dmitry-timofeev) [reply .travis.yml]: I see, thanks! So we could remove that line and have exonum pick up the system one?

Comment (dmitry-timofeev) [reply .travis.yml]: Updated and cleared the caches on Travis to get a clean build, will see how it goes: https://travis-ci.org/exonum/exonum-java-binding/builds/411277584

**Pull request merged**